Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -0,0 +1,5870 @@
+    };
+
+    template<typename... Args>
+    void logInstructionData(bool first, const char* const& literal, const Args&... args)
+    {
+        if (!first)
+            dataLog(" ");
+
+        dataLog(literal);
+        if (!std::strcmp(literal, "=> "))
+            logInstructionData(true, args...);
+        else
+            logInstructionData(false, args...);
+    }
+
+    template<typename T, typename... Args>
+    void logInstructionData(bool first, const Result<T>& result, const Args&... args)
+    {
+        if (!first)
+            dataLog(" ");
+
+        dataLog("=> ");
+        logInstructionData(true, result.value, args...);
+    }
+
+    template<typename T, typename... Args>
+    void logInstructionData(bool first, const T& t, const Args&... args)
+    {
+        if (!first)
+            dataLog(", ");
+        dataLog(t);
+        logInstructionData(false, args...);
+    }
+
+#define RESULT(...) Result { __VA_ARGS__ }
+#define LOG_INSTRUCTION(...) do { if (UNLIKELY(Options::verboseBBQJITInstructions())) { logInstruction(__VA_ARGS__); } } while (false)
+#define LOG_INDENT() do { if (UNLIKELY(Options::verboseBBQJITInstructions())) { m_loggingIndent += 2; } } while (false);
+#define LOG_DEDENT() do { if (UNLIKELY(Options::verboseBBQJITInstructions())) { m_loggingIndent -= 2; } } while (false);
+
+public:
+    static constexpr bool tierSupportsSIMD = false; // TODO: Support SIMD
+
+    BBQJIT(CCallHelpers& jit, const TypeDefinition& signature, Callee& callee, const FunctionData& function, uint32_t functionIndex, const ModuleInformation& info, Vector<UnlinkedWasmToWasmCall>& unlinkedWasmToWasmCalls, MemoryMode mode, InternalFunction* compilation, TierUpCount* tierUp)
+        : m_jit(jit)
+        , m_callee(callee)
+        , m_function(function)
+        , m_functionSignature(signature.as<FunctionSignature>())
+        , m_functionIndex(functionIndex)
+        , m_info(info)
+        , m_mode(mode)
+        , m_unlinkedWasmToWasmCalls(unlinkedWasmToWasmCalls)
+        , m_tierUp(tierUp)
+        , m_gprBindings(jit.numberOfRegisters(), RegisterBinding::none())
+        , m_fprBindings(jit.numberOfFPRegisters(), RegisterBinding::none())
+        , m_gprLRU(jit.numberOfRegisters())
+        , m_fprLRU(jit.numberOfFPRegisters())
+        , m_lastUseTimestamp(0)
+        , m_compilation(compilation)
+    {
+        RegisterSetBuilder gprSetBuilder = RegisterSetBuilder::allGPRs();
+        gprSetBuilder.exclude(RegisterSetBuilder::specialRegisters());
+
+        m_memoryBaseGPR = GPRInfo::wasmBaseMemoryPointer;
+        gprSetBuilder.remove(m_memoryBaseGPR);
+
+        m_wasmContextInstanceGPR = GPRInfo::wasmContextInstancePointer;
+        if (m_wasmContextInstanceGPR != InvalidGPRReg)
+            gprSetBuilder.remove(m_wasmContextInstanceGPR);
+
+        m_boundsCheckingSizeGPR = GPRInfo::wasmBoundsCheckingSizeRegister;
+        if (mode == MemoryMode::BoundsChecking)
+            gprSetBuilder.remove(m_boundsCheckingSizeGPR);
```

Comment:
Let's just use GPRInfo::wasmBoundsCheckingSizeRegister etc. directly.

Reply:
Resolved in latest push.